### PR TITLE
Imprved the checking for BP Class Plugin

### PR DESCRIPTION
**Pull request by narenin**
Trac ticket: https://buddypress.trac.wordpress.org/ticket/9181

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

File: src/bp-templates/bp-nouveau/buddypress-functions.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ function() {
 			);
 
 			// When BP Classic is activated, regular themes need this filter. 
-			if ( function_exists( 'bp_classic' ) ) {
+			if ( function_exists( 'bp_classic' ) || is_plugin_active( 'bp-classic/bp-classic.php' ) || is_plugin_active( 'buddyclassic/bp-classic.php' ) ) {
```
Comment (renatonascalves) [inline src/bp-templates/bp-nouveau/buddypress-functions.php:125]: Is `buddyclassic` a real plugin?

Comment (narenin) [reply src/bp-templates/bp-nouveau/buddypress-functions.php:125]: @renatonascalves Thanks for reviewing this PR. I have added this just for the case when user will rename folder to `buddyclassic` from `bp-classic`

Comment (narenin) [reply src/bp-templates/bp-nouveau/buddypress-functions.php:125]: @renatonascalves Please let me know if I need to made any other changes in this.

Comment (renatonascalves) [reply src/bp-templates/bp-nouveau/buddypress-functions.php:125]: I'd remove the buddyclassic part, actually. It is not a _real_ plugin.

Comment (renatonascalves) [inline src/bp-templates/bp-nouveau/buddypress-functions.php:125]: There are many places where we are doing similar checks that we should update too: https://github.com/search?q=org%3Abuddypress+function_exists(+%27bp_classic%27+)&type=Code

Maybe create a new function to check if the BP Classic plugin is active?! And reuse it where applicable.

**Review by renatonascalves — CHANGES_REQUESTED** [review]

File: src/bp-templates/bp-nouveau/buddypress-functions.php (hunk 1 above)
Comment (narenin) [reply src/bp-templates/bp-nouveau/buddypress-functions.php]: @renatonascalves Thanks for the detailed feedback, I have implemented the suggestions can you please take a look.

Comment (narenin) [reply src/bp-templates/bp-nouveau/buddypress-functions.php]: @renatonascalves @imath Please let me know if there is any feedback for me.

File: src/bp-core/bp-core-functions.php [hunk 2]
```diff
@@ -5286,3 +5286,17 @@ function bp_is_admin( $screen_id = '' ) {
...
+ *
+ * @since 15.0.0
+ *
+ * @return bool True if bp_classic is active, false otherwise.
+ */
+function is_bp_classic_active() {
+    if ( function_exists( 'bp_classic' ) || is_plugin_active( 'bp-classic/bp-classic.php' ) ) {
+        return true;
+    }
+    return false;
```
Comment (imath) [inline src/bp-core/bp-core-functions.php:5301]:
```suggestion
    return function_exists( 'bp_classic' ) || is_plugin_active( 'bp-classic/bp-classic.php' );
```
I believe we can save 2 lines of code 😅

**Review by imath — COMMENTED** [review]
Hi @narenin thanks for your contribution.

I wonder what's fastest, if checking if BP Classic is active using the WordPress function and since this seems required we could probably remove the `function_exists` check?

File: src/bp-core/bp-core-functions.php (hunk 2 above)
Comment (narenin) [reply src/bp-core/bp-core-functions.php]: @imath Thanks for reviewing this. I have implemented the suggestions please check.

File: src/bp-core/bp-core-functions.php [hunk 3]
```diff
@@ -5286,3 +5286,14 @@ function bp_is_admin( $screen_id = '' ) {
...
 }
+
+/**
+ * Common function to check if bp_classic is active or not.
+ *
+ * @since 15.0.0
+ *
+ * @return bool True if bp_classic is active, false otherwise.
+ */
+function is_bp_classic_active() {
```
Comment (renatonascalves) [inline src/bp-core/bp-core-functions.php:5297]: One last suggestion to start with our custom prefix: `bp_`
```suggestion
function bp_is_classic() {
```

**Review by renatonascalves — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-functions.php (hunk 3 above)
Comment (narendraie) [reply src/bp-core/bp-core-functions.php]: @renatonascalves Thanks for the suggestion, I have implemented the same please check.

Comment (narenin) [reply src/bp-core/bp-core-functions.php]: @renatonascalves Please let me know if any other changes needs to be made.

**Review by renatonascalves — APPROVED** [review]
All good! 🎈

File: src/bp-core/bp-core-functions.php [hunk 4]
```diff
@@ -5286,3 +5286,14 @@ function bp_is_admin( $screen_id = '' ) {
...
+
+/**
+ * Common function to check if bp_classic is active or not.
+ *
+ * @since 15.0.0
+ *
+ * @return bool True if bp_classic is active, false otherwise.
+ */
+function bp_is_classic() {
+	return function_exists( 'bp_classic' ) || is_plugin_active( 'bp-classic/bp-classic.php' );
```
Comment (imath) [inline src/bp-core/bp-core-functions.php:5298]: All PHP Unit tests are failing, we need more work about this PR. For a reason I ignore for now the `is_plugin_active()` function is not yet available when doing these tests.

`PHP Fatal error:  Uncaught Error: Call to undefined function is_plugin_active() in /tmp/wordpress/wp-content/plugins/buddypress/src/bp-core/bp-core-functions.php:5298`

**Review by imath — CHANGES_REQUESTED** [review]
PHP Unit tests are not good 😓 , let's understand why!

File: src/bp-core/bp-core-functions.php (hunk 4 above)
Comment (narenin) [reply src/bp-core/bp-core-functions.php:5298]: @imath Thanks for the review and feedback, should we add something like this to `tests/phpunit/bootstrap.php` to handle that fatal error?

`if ( ! function_exists( 'is_plugin_active' ) ) {
    require_once ABSPATH . 'wp-admin/includes/plugin.php';
}
`

Comment (renatonascalves) [reply src/bp-core/bp-core-functions.php:5298]: I _think_ `is_plugin_active` is not loaded for front-end requests, only `wp-admin` requests. So when we load `bp-nouveau`, we make use of this function and it is complaining it doesn't exist.

```php
if ( ! function_exists( 'is_plugin_active' ) ) {
  require_once ABSPATH . 'wp-admin/includes/plugin.php';
} 
```

^ this seems to be the solution suggested [here](https://developer.wordpress.org/reference/functions/is_plugin_active/#comment-616) for front-end support too.

File: src/bp-core/bp-core-functions.php [hunk 5]
```diff
@@ -5286,3 +5286,14 @@ function bp_is_admin( $screen_id = '' ) {
...
+
+/**
+ * Common function to check if bp_classic is active or not.
+ *
+ * @since 15.0.0
+ *
+ * @return bool True if bp_classic is active, false otherwise.
+ */
+function bp_is_classic() {
+	return function_exists( 'bp_classic' ) || ( function_exists( 'is_plugin_active' ) && is_plugin_active( 'bp-classic/bp-classic.php' ) );
```
Comment (renatonascalves) [inline src/bp-core/bp-core-functions.php:5298]: I'm under the impression that @imath suggested to add `is_plugin_active` to this file. Like a polyfill. 🤔 

https://github.com/buddypress/buddypress/blob/master/src/bp-core/bp-core-wpabstraction.php#L190%7CL203

Although, your approach works too. So I guess the question is if we wanna polyfill `is_plugin_active `

**Review by renatonascalves — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-functions.php (hunk 5 above)
Comment (narenin) [reply src/bp-core/bp-core-functions.php:5298]: @imath @renatonascalves Thanks for the feedback, I have implemented the feedback, could you please take the look.

**Pull request closed without merging**